stock_picking_auto_create_lot_qty/models/stock_move_line.py

```diff
@@ -0,0 +1,35 @@
+# Copyright (C) 2023 Cetmix OÜ
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+from odoo import models
+from odoo.fields import first
+
+
+class StockMoveLine(models.Model):
+
+    _inherit = "stock.move.line"
+
+    def set_lot_auto(self):
+        """
+        Create lots using create_multi to avoid too much queries
+        As move lines were created by product or by tracked 'serial'
+        products, we apply the lot with both different approaches.
+        """
+        values = []
+        production_lot_obj = self.env["stock.production.lot"]
+        lots_by_product = dict()
+        for line in self:
+            values.append(line._prepare_auto_lot_values())
+        lots = production_lot_obj.create(values)
+        for lot in lots:
+            if lot.product_id.id not in lots_by_product:
+                lots_by_product[lot.product_id.id] = lot
+            else:
+                lots_by_product[lot.product_id.id] += lot
+        for line in self:
+            lot = first(lots_by_product[line.product_id.id])
+            lots_by_product[line.product_id.id] = (
+                lots_by_product[line.product_id.id] - lot
+            )
+            line.lot_id = lot
+            if lot.product_id.tracking == "serial":
+                lots_by_product[line.product_id.id] -= lot
```

Comment:
This is almost the same as `stock_picking_auto_create_lot` except for lines 30-32.

Is there a reason I'm not seeing for not running super() and adding only these lines at the end?

Reply:
If I do as you say, we will get following situation:
- in super will created lots
- for all stock move lines will be assigned only one lot, other lots will be not assgned to any stock move line
- i can not find reliable way to find other lots that are not assigned to stock move line wich are created in super, may be you can give me a tip with it.

Reply:
The more I think about it the more I think we can probably just fix this function in the base module directly. I've also done some light testing of this change, and it seems fine. It also doesn't break any automated tests.

When `_set_auto_lot` is executed, we already have the appropriate `stock.move.line` objects (split in individual objects in case they're tracked by serial. So really what we want to do is just assign one lot to each line, in order.

In a separate commit, you can change these two lines by removing the if (and indenting appropriately)

https://github.com/OCA/stock-logistics-workflow/blob/14.0/stock_picking_auto_create_lot/models/stock_move_line.py#L38-L39

This should allow you to completely remove this entire file in your PR.

Reply:
done